src/bitbucket/__tests__/BitbucketPipelinesAPI.test.ts

```diff
@@ -0,0 +1,37 @@
+import axios from 'axios';
+import { BitbucketPipelinesAPI } from '../BitbucketPipelinesAPI';
+
+jest.mock('axios');
+const mockedAxios = axios as unknown as jest.Mocked<typeof axios>;
+
+jest.mock('../BitbucketAuthenticator', () => ({
+  bitbucketAuthenticator: {
+    getAuthConfig: () => Promise.resolve({}),
```

Comment:
generally a good practice to use `jest.fn()` so the mock implementation can be changed between tests.

e.g.

```
getAuthConfig: jest.fn(() => Promise.resolve({}))
```

Can also use make the fn `async` to get rid of the `Promise.resolve`

Reply:
Changed the mocking to use `mockResolvedValue`.